lib/Sema/CSRanking.cpp

```diff
@@ -766,6 +766,67 @@ static void addKeyPathDynamicMemberOverloads(
   }
 }
 
+/// In certain cases we want to compare the type bindings chosen for different
+/// overloads. This function adds such type bindings to \p typeDiff. This is
+/// currently only done for short-form and delegating constructor calls.
+static void
+addOverloadBindingTypeDiffs(ConstraintSystem &cs, ConstraintLocator *loc,
+                            ArrayRef<Solution> solutions, unsigned idx1,
+                            unsigned idx2,
+                            SmallVectorImpl<std::pair<Type, Type>> &typeDiff) {
```

Comment:
Alternatively, could we mark locator for `methodTy` as related to `TypeChecker::getSelfForInitDelegationInConstructor` and extract a an argument tuple from it during "compare type bindings" phase? It seems like that way we don't have to replicate the logic from `matchTupleTypes` here because it would be handled by `isSubtypeOf` call.